pkg/cmd/completion.go

```diff
@@ -15,24 +15,39 @@
 package cmd
 
 import (
+	"os"
+
 	"github.com/spf13/cobra"
 )
 
 // completionCmd represents the completion command
 var completionCmd = &cobra.Command{
-	Use:   "completion",
-	Short: "Generate bash-completion file\n",
+	Use:   "completion <bash|zsh>",
+	Short: "Generate bash or zsh completion script\n",
+	Long:  ``,
+}
+
+var bashCompletionCmd = &cobra.Command{
+	Use:   "bash",
+	Short: "Generate bash completion script",
 	Long:  ``,
 	Run: func(cmd *cobra.Command, args []string) {
-		generateBashCompletion()
+		err := RootCmd.GenBashCompletion(os.Stdout)
+		checkError(err)
 	},
 }
 
-func init() {
+var zshCompletionCmd = &cobra.Command{
+	Use:   "zsh",
+	Short: "Generate zsh completion script",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
You could prefer the usage of RunE and return err. `checkError` and `os.Exit` is something that break the testability.